tests/imagetools.go

```diff
@@ -14,25 +15,15 @@ import (
 
 var imagetoolsTests = []func(t *testing.T, sb integration.Sandbox){
 	testImagetoolsInspectAndFilter,
+	testImagetoolsAnnotation,
 }
 
 func testImagetoolsInspectAndFilter(t *testing.T, sb integration.Sandbox) {
 	if sb.Name() != "docker-container" {
 		t.Skip("imagetools tests are not driver specific and only run on docker-container")
 	}
 
-	dockerfile := []byte(`
-	FROM scratch
-	ARG TARGETARCH
-	COPY foo-${TARGETARCH} /foo
-	`)
-	dir := tmpdir(
-		t,
-		fstest.CreateFile("Dockerfile", dockerfile, 0600),
-		fstest.CreateFile("foo-amd64", []byte("foo-amd64"), 0600),
-		fstest.CreateFile("foo-arm64", []byte("foo-arm64"), 0600),
-	)
-
+	dir := createDockerfile(t)
```

Comment:
We shouldn't split this out to a separate function, this is specific to doing multi-arch filtering stuff. We can define a much simpler dockerfile for the annotations tests IMO.

Reply:
I don't have a strong opinion on this so would be fine going to previous/different implementation. But it has been suggested [here](https://github.com/docker/buildx/pull/1965#discussion_r1281631431).

Reply:
Ah fair, I missed that, sorry!